maven-plugin/src/main/scala/akka/grpc/maven/GenerateMojo.scala

```diff
@@ -28,76 +66,109 @@ class GenerateMojo @Inject() (project: MavenProject) extends AbstractMojo {
   var generateServer: Boolean = _
 
   override def execute(): Unit = {
-    val protoDirectory = new File(project.getBasedir, protoPath).getAbsoluteFile
-    if (!protoDirectory.exists()) throw new RuntimeException(s"Configured protoPath [$protoDirectory] does not exist")
-    val schemas = findProtoFiles(protoDirectory)
-
-    val sourceRoot = "target/generated-sources/akka-grpc-" + language.name().toLowerCase
-    val sourceManaged = new File(sourceRoot)
-    project.addCompileSourceRoot(sourceRoot)
-
-    val akkaGrpcCodeGeneratorSettings = Seq("flat_package")
-    val targets = language match {
-      case Language.JAVA ⇒
-        val glueGenerator =
-          if (generateServer)
-            if (generateClient) JavaBothCodeGenerator
-            else JavaServerCodeGenerator
-          else JavaClientCodeGenerator
-        Seq[Target](
-          protocbridge.gens.java -> sourceManaged,
-          adaptAkkaGenerator(sourceManaged, glueGenerator, akkaGrpcCodeGeneratorSettings))
-      case Language.SCALA ⇒
-        val glueGenerator =
-          if (generateServer)
-            if (generateClient) ScalaBothCodeGenerator
-            else ScalaServerCodeGenerator
-          else ScalaClientCodeGenerator
-        Seq[Target](
-          (JvmGenerator("scala", ScalaPbCodeGenerator), akkaGrpcCodeGeneratorSettings) → sourceManaged,
-          adaptAkkaGenerator(sourceManaged, glueGenerator, akkaGrpcCodeGeneratorSettings))
+
+    // generated sources should be compiled
+    val generatedSourcesDir = "target/generated-sources/akka-grpc-" + language.name().toLowerCase
+    val compileSourceRoot = new File(project.getBasedir, generatedSourcesDir)
+    project.addCompileSourceRoot(generatedSourcesDir)
+
+
+
+    // only do any work if needed (m2e integration)
+    if (!buildContext.isIncremental && buildContext.hasDelta(generatedSourcesDir)) {
```

Comment:
This does not make intuitive sense to me, but I'm not familiar with how this works so it might be OK?

Reply:
Yeah, the reason todo it is that eclipse has two phases, one for config and one for actually building. This was pretty much copied ad lib from one of the Java WSDL-generating plugins.

We don't actually do much incremental support as that would require hooking into the compilation in more fine grained detail - keeping track of what .proto generated what files etc.

Reply:
I wonder if it should really be generatedSourcesDir though

Reply:
Actually, looking at a different generator-style maven plugin (antlr) I see they don't do this, let's skip this condition and always try to compile.